### DISCUSSION TITLE
Reclaiming --audit functionality

### DISCUSSION THREAD

**Discussion by emerajid** [netblue30/firejail#5997] [Q&A]
Previously, it was possible to use --audit flag to review your sandbox for basic flaws, which was a very useful feature. Now, however, this option is gone. Instead of it I see jailcheck utility which is terribly useless, because requires you to have access to sudo. I am not going to allow sudo. I am not going to allow su either. firejail is enough of security issues already.

How can I get --audit option back?

**Comment by rusty-snake** [comment]
> How can I get --audit option back?

Patch and compile.

Or grab the faudit code, build the binary and start it like `firejail --profile=PROFILE faudit`.

Also note that audit was never more that a false feeling of security.

Self marketing: Or use [raudit](https://github.com/rusty-snake/raudit).

> because requires you to have access to sudo. I am not going to allow sudo. I am not going to allow su either.

It requires root not sudo.

How do you install updates for example? Second user with sudo?

> firejail is enough of security issues already.

https://github.com/igo95862/bubblejail
https://codeberg.org/crabjail/crablock

**Comment by emerajid** [comment]
>Patch and compile.

Not a programmer, so alas!

>Or grab the faudit code

Thank you.

>Also note that audit was never more that a false feeling of security.

https://madaidans-insecurities.github.io/linux.html
The whole Linux thing is a huge false sense of security. I didn't use faudit for security, I used it to simply check if I didn't miss some trivial things and in that it was very useful.

>Self marketing

Is raudit better? If it is I will use it.

>It requires root not sudo.

I tried running it as root on home directory of target user and it refused.

>Second user with sudo?

This. Only one user has access to the sudo/su thing. I however must admit that this is paranoid bullshit for systemd systems (my case), because there is polkit and it is deliberately designed in a way no normal user will be able to configure it ever. And it by default allows anyone to authenticate as root, for example through 'systemctl edit' for crafting a malicious service. Paranoia is so paranoia.

I will look into your links maybe, but I already have some working firejail configs and I am not very keen to rewrite them again.

**Reply by rusty-snake** [reply]
> Is raudit better?

Better can be subjective. It uses configurable rules rather than hardcoded rules. The approach might be better. However I only implemented four rule types.
